File: src/main/java/com/thunderbear06/ai/task/TaskManager.java
package com.thunderbear06.ai.task;

public class TaskManager {
    private Task currentTask;

    public void setCurrentTask(Task task) {
        clearCurrentTask();
        currentTask = task;
        currentTask.firstTick();
    }

    public String getCurrentTaskName() {
        return currentTask == null ? "idle" : currentTask.getName();
    }

    public void clearCurrentTask() {
        if (currentTask == null) return;
        currentTask.lastTick();
        currentTask = null;
    }

    public void tick() {
        if (currentTask == null) return;
        if (currentTask.shouldTick()) currentTask.tick();
        else clearCurrentTask();
    }

    public boolean isIdle() {
        return currentTask == null;
    }
}

File: src/main/kotlin/com/thunderbear06/ai/task/TaskManager.kt
package com.thunderbear06.ai.task

class TaskManager {
    private var currentTask: Task? = null

    fun setCurrentTask(task: Task) {
        clearCurrentTask()
        currentTask = task
        task.firstTick()
    }

    fun getCurrentTaskName(): String = currentTask?.name ?: "idle"

    fun clearCurrentTask() {
        val task = currentTask ?: return
        task.lastTick()
        currentTask = null
    }

    fun tick() {
        val task = currentTask ?: return
        if (task.shouldTick()) task.tick() else clearCurrentTask()
    }

    fun isIdle(): Boolean = currentTask == null
}
